### PR TITLE
Added error messages for everything under ECS folder

### PR DESCRIPTION
**Pull request by NullSapphire**
Added error messages for every exception under the ECS folder that was missing an error message, and made some new messages that weren't present previously.

### CODE REVIEW

File: src/NovelRT.Interop/NrtInteropErrorHandling.cpp [hunk 1]
```diff
@@ -3,19 +3,25 @@
...
-const char* const errMsgIsArgumentOutOfRange = "Unable to continue! The argument is out of range.";
-const char* const errMsgIsInvalidOperation = "Unable to continue! The current operation is invalid.";
-const char* const errMsgIsCharacterNotFound = "Unable to continue! The character can't be found.";
-const char* const errMsgIsOutOfMemory = "Insufficient memory to continue the execution of the program.";
+const char* const errMsgIsNullInstanceProvided = "Unable to continue. A null instance was provided when an instance was expected.";
+const char* const errMsgIsNaN = "Unable to continue. Calculation resulted in an object that is not a number (NaN).";
+const char* const errMsgIsDivideByZero = "Unable to continue. Divide by zero was attempted.";
+const char* const errMsgIsNotSupportedInternal = "Unable to continue. The current operation is not supported.";
+const char* const errMsgDupKeyGiven = "Unable to continue. Duplicate key given.";
+const char* const errMsgCompilationError = "Unable to continue. Compilation error.";
```
Comment (RubyNova) [inline src/NovelRT.Interop/NrtInteropErrorHandling.cpp:11]: What error is this for?

File: include/NovelRT.Interop/NrtErrorHandling.h [hunk 2]
```diff
@@ -27,6 +31,9 @@ extern "C"
     void Nrt_setErrMsgIsInvalidOperationInternal();
     void Nrt_setErrMsgIsCharacterNotFoundInternal();
     void Nrt_setErrMsgIsOutOfMemoryInternal();
+    void Nrt_setErrMsgKeyNotFound();
+    void Nrt_setErrMsgRuntimeNotFound();
+    void Nrt_setErrMsgErrorUnknown();
```
Comment (RubyNova) [inline include/NovelRT.Interop/NrtErrorHandling.h:36]: Same here for the `Internal` suffix:
```suggestion
    void Nrt_setErrMsgKeyNotFoundInternal();
    void Nrt_setErrMsgRuntimeNotFoundInternal();
    void Nrt_setErrMsgErrorUnknownInternal();
```

**Review by RubyNova — CHANGES_REQUESTED** [review]
Good start for a first pass, just some conventional stuff from me - and obviously you'll need to make it pass CI too.

File: src/NovelRT.Interop/NrtInteropErrorHandling.cpp (hunk 1 above)
Comment (capnkenny) [reply src/NovelRT.Interop/NrtInteropErrorHandling.cpp:11]: Which are you calling out here? `errMsgCompilationError`? GH preview shows 4 different messages so just making sure the question is clear

Comment (RubyNova) [reply src/NovelRT.Interop/NrtInteropErrorHandling.cpp:11]: I clicked the last one, not sure why its done that. Yes I mean this.

**Review by RubyNova — APPROVED** [review]
LGTM

**Pull request merged**